src/frontend/src/btc/utils/btc-transactions.utils.ts

```diff
@@ -0,0 +1,29 @@
+import type { BtcTransactionUi } from '$btc/types/btc';
+import type { BtcAddress } from '$lib/types/address';
+import type { BitcoinOutput, BitcoinTransaction } from '$lib/types/blockchain';
+import { nonNullish } from '@dfinity/utils';
+
+export const mapBtcTransaction = ({
+	transaction: { inputs, block_index, out, hash, time },
+	btcAddress
+}: {
+	transaction: BitcoinTransaction;
+	btcAddress: BtcAddress;
+}): BtcTransactionUi => {
+	const isTypeSend = inputs.some(({ prev_out }) => prev_out.addr === btcAddress);
+	const blockIndexAvailable = nonNullish(block_index);
+	const output: BitcoinOutput | undefined = out.find(({ addr }) =>
+		isTypeSend ? addr !== btcAddress : addr === btcAddress
+	);
+
+	return {
+		id: hash,
+		timestamp: time,
+		value: BigInt(output?.value ?? 0),
+		status: blockIndexAvailable ? 'confirmed' : 'pending',
+		blockNumber: block_index ?? undefined,
+		type: isTypeSend ? 'send' : 'receive',
+		from: isTypeSend ? btcAddress : inputs[0].prev_out.addr,
```

Comment:
`inputs[0]` and `inputs[0].prev_out` are never undefined?

Reply:
According to the API, those values should always be there.